node/src/components/storage.rs

```diff
@@ -1021,11 +1069,110 @@ impl Storage {
             return Err(LmdbExtError::BlockHeaderNotStoredUnderItsHash {
                 queried_block_hash: *block_hash,
                 found_block_header_hash,
+                block_header: Box::new(block_header),
             });
         };
         Ok(Some(block_header))
     }
 
+    /// Retrieves a single Merklized block body in a separate transaction from storage.
+    fn get_single_v2_block_body<Tx: Transaction>(
+        &self,
+        tx: &mut Tx,
+        block_body_hash: &Digest,
+    ) -> Result<Option<BlockBody>, LmdbExtError> {
+        let [hashed_deploy_hashes, hashed_transfer_hashes, hashed_proposer]: [Digest;
+            BlockBodyHashedParts::NUMBER_OF_HASHES] =
+            match tx.get_value(self.block_body_v2_db, block_body_hash)? {
+                Some(parts_hashes) => parts_hashes,
+                None => return Ok(None),
+            };
+
+        let deploy_hashes: Vec<DeployHash> =
+            match tx.get_value(self.deploy_hashes_db, &hashed_deploy_hashes)? {
+                Some(deploy_hashes) => deploy_hashes,
+                None => return Ok(None),
```

Comment:
Yes.

But in the long run we won't always have the proposer, however.  If a block gains more fields then we'll only have deploys and transfers.

Reply:
If we _add_ fields, it will still have a proposer, will it?

Reply:
@afck Based on @fizyk20's [suggestion below][1] I've updated the scheme in https://github.com/casper-network/casper-node/commit/7388d0d0c843d82ca38d18d7c8a03b7860af7fd6 so that all blocks are stored as Merklized linked lists.

[1]: https://github.com/casper-network/casper-node/pull/1563#discussion_r656187963